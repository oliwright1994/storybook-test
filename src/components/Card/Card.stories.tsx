import { Text } from '@chakra-ui/core'
import React from 'react'

import { MessageType } from '../../constants'
import createStory from '../../storybook/createStory'
import Tag from '../Tag'
import Card from './Card'

createStory('COMPONENTS | Card', [
  {
    label: 'Card',
    jsx: () => {
      return (
        <Card>
          <Text>This is Text component inside Card component</Text>
          <Tag messageType={MessageType.success}>This is Tag component inside Card component</Tag>
        </Card>
      )
    },
  },
])
