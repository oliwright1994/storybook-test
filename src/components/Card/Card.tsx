import { Box } from '@chakra-ui/core'
import React, { FC } from 'react'

import { Space } from '../../constants'

const Card: FC = ({ children }) => {
  return (
    <Box p={Space.l} border="1px" color="grey">
      {children}
    </Box>
  )
}

export default Card
