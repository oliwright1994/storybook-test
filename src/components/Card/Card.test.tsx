import { Text } from '@chakra-ui/core'
import { screen } from '@testing-library/react'
import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Card from '.'

describe('<Card />', () => {
  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(
      <Card>
        <Text>Test One</Text>
        <Text>Test Two</Text>
        <Text>Test Three</Text>
      </Card>
    )
    const textNodes = screen.getAllByText(/test/i)

    expect(textNodes).toHaveLength(3)
    expect(container).toMatchSnapshot()
  })
})
