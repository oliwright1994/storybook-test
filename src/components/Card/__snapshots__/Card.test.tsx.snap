// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render as expected 1`] = `
<div>
  <div
    class="css-4iz7qw"
  >
    <p
      class="css-fzcsno"
    >
      Test One
    </p>
    <p
      class="css-fzcsno"
    >
      Test Two
    </p>
    <p
      class="css-fzcsno"
    >
      Test Three
    </p>
  </div>
</div>
`;
